src/lib/genesis_ledger/genesis_ledger.ml

```diff
@@ -99,29 +129,37 @@ include Utils
 
 module Make (Inputs : Intf.Ledger_input_intf) : Intf.S = struct
   include Inputs
+  include Backing
 
   (* TODO: #1488 compute this at compile time instead of lazily *)
-  let t =
+  let backing_ledger =
     let open Lazy.Let_syntax in
     let%map ledger, insert_accounts =
       match directory with
       | `Ephemeral ->
-          lazy (Ledger.create_ephemeral ~depth (), true)
+          lazy (`Ephemeral (Ledger.create_ephemeral ~depth ()), true)
       | `New ->
-          lazy (Ledger.create ~depth (), true)
+          lazy (`Root (Ledger.Root.create_single ~depth ()), true)
       | `Path directory_name ->
-          lazy (Ledger.create ~directory_name ~depth (), false)
+          lazy
+            (`Root (Ledger.Root.create_single ~directory_name ~depth ()), false)
     in
     ( if insert_accounts then
+      let mask = as_masked ledger in
       let addrs_and_accounts =
-        let ledger_depth = Ledger.depth ledger in
+        let ledger_depth = Ledger.depth mask in
         Lazy.force accounts
         |> List.mapi ~f:(fun i (_, acct) ->
                (Ledger.Addr.of_int_exn ~ledger_depth i, acct) )
       in
-      Ledger.set_batch_accounts ledger addrs_and_accounts ) ;
+      Ledger.set_batch_accounts mask addrs_and_accounts ) ;
```

Comment:
I think this is it - this creates a mask, populates it with accounts, and then forgets about the mask entirely. Nothing gets stored in the underlying database at all.

This is actually interesting! The old code also called `set_batch_accounts` on the masked `Ledger.t`, so I think it was actually completely irrelevant that the `New` ledger was created as a database-backed ledger in the old code.

Reply:
Actually, it's probably safer to `set_batch_accounts` on the database itself, if possible, just in case someone tries to populate a root ledger with a genesis ledger before the anyone commits the genesis ledger mask.

Reply:
Even better would be to have the exposed genesis ledger be an `Any_ledger.witness`. I'll try that.

Reply:
No, that's a little invasive. I've kept it as it is now - I just store the mask and commit it before doing anything with the backing `Root`.